### ISSUE TITLE
[Test Failure] FSInternetRadioHandlerOSGiTest

### ISSUE THREAD

**Issue by sjka** [eclipse/smarthome#3223]
In a [Travis CI build](https://travis-ci.org/eclipse/smarthome/builds/220120289) of #3222, there was a weird problem in the `verify the volume is updated through the CHANNEL_VOLUME_PERCENT` of `FSInternetRadioHandlerOSGiTest`:

```
verify the volume is updated through the CHANNEL_VOLUME_PERCENT(org.eclipse.smarthome.binding.fsinternetradio.test.FSInternetRadioHandlerOSGiTest)  Time elapsed: 11.163 sec  <<< FAILURE!
java.lang.AssertionError: The item's state was not updated correctly in attempt to use the INCREASE command
Expected: is <1>
     but: was <2>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:865)
	at org.junit.Assert$assertThat$0.callStatic(Unknown Source)
	at org.eclipse.smarthome.binding.fsinternetradio.test.FSInternetRadioHandlerOSGiTest$_testChannelWithINCREASEAndDECREASECommands_closure24.doCall(FSInternetRadioHandlerOSGiTest.groovy:498)
	at org.eclipse.smarthome.binding.fsinternetradio.test.FSInternetRadioHandlerOSGiTest$_testChannelWithINCREASEAndDECREASECommands_closure24.doCall(FSInternetRadioHandlerOSGiTest.groovy)
	at sun.reflect.GeneratedMethodAccessor12.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:278)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1016)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:39)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:112)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:218)
	at sun.reflect.GeneratedMethodAccessor8.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:153)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:189)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy)
	at sun.reflect.GeneratedMethodAccessor7.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:133)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:141)
	at org.eclipse.smarthome.binding.fsinternetradio.test.FSInternetRadioHandlerOSGiTest.testChannelWithINCREASEAndDECREASECommands(FSInternetRadioHandlerOSGiTest.groovy:497)
```

I can't really make sense of this on a first glance, but since the problem vanished on the second build of the very same commit it seems like the test is not stable.

**Comment by sjka** [comment]
@m-memova, as you know these tests best, does anything cross your mind which could cause this?

**Comment by m-memova** [comment]
@sjka , I will take a look as soon as I can. As far as I remember the problem with these tests was that changing the item's status several times in one test case could not always be handled quick enough.

**Comment by m-memova** [comment]
So the main issue was that the volume could be changed through two different channels. And there were several test cases changing it. Every volume change is related to sending a HTTP request which is asynchronous by its nature. I think that was the reason for the inconsistent behavior of the tests. Once in 200-300 times they could fail because of a thread concurrency issue. I made 2 changes in order to fix that:
1) I made the volume tests a little bit more simple, extracting part of the code into new test cases.
2) I synchronized the two common methods on the channel uid. 
I ran the tests hundreds of times in a row on my machine (with Eclispe and Maven) and everything seems fine.
A PR is created : https://github.com/eclipse/smarthome/pull/3266
